neuraxle/metaopt/random.py

```diff
@@ -25,27 +25,264 @@
+            'ValidationSplitWrapper', DataContainer):
+        """
+        Fit using the training split.
+        Calculate the scores using the validation split.
+
+        :param context: execution context
+        :param data_container: data container
+        :type context: ExecutionContext
+        :type data_container: DataContainer
+        :return: fitted self
+        """
+        new_self, results_data_container = self.handle_fit_transform(data_container, context)
+        return new_self, data_container
+
+    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext) -> (
+            'BaseStep', DataContainer):
+        """
+        Fit Transform given data inputs without splitting.
+
+        :param context:
+        :param data_container: DataContainer
+        :type data_container: DataContainer
+        :type context: ExecutionContext
+        :return: outputs
+        """
+        train_data_container, validation_data_container = self.split_data_container(data_container)
+
+        self.wrapped, _ = self.wrapped.handle_fit(train_data_container, context.push(self.wrapped))
+
+        results_data_container = self.wrapped.handle_transform(train_data_container, context.push(self.wrapped))
+
+        self._update_scores_train(results_data_container.data_inputs, results_data_container.expected_outputs)
+
+        if self.run_validation_split_in_test_mode:
+            self.set_train(False)
+
+        results_data_container = self.wrapped.handle_transform(validation_data_container, context.push(self.wrapped))
+
+        self.set_train(True)
+
+        self._update_scores_validation(results_data_container.data_inputs, results_data_container.expected_outputs)
+
+        data_container = self.wrapped.handle_transform(data_container, context.push(self.wrapped))
+
+        return self, data_container
+
+    def handle_transform(self, data_container: DataContainer, context: ExecutionContext):
+        """
+        Transform given data inputs without splitting.
+
+        :param context: execution context
+        :param data_container: DataContainer
+        :type data_container: DataContainer
+        :type context: ExecutionContext
+        :return: outputs
+        """
+        return self.wrapped.handle_transform(data_container, context.push(self.wrapped))
+
+    def fit(self, data_inputs, expected_outputs=None) -> 'ValidationSplitWrapper':
+        """
+        Fit using the training split.
+        Calculate the scores using the validation split.
+
+        :param data_inputs: data inputs
+        :param expected_outputs: expected outputs
+        :return: fitted self
+        """
+        train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs = self.split(
+            data_inputs, expected_outputs)
+
+        self.wrapped = self.wrapped.fit(train_data_inputs, train_expected_outputs)
+
+        train_data_inputs = self.transform(train_data_inputs)
```

Comment:
Misleading name. Rename the return value `train_data_inputs` to `train_predicted_outputs`.

Reply:
Also, don't do a `self.transform` here. Use `self.wrapped.predict()` instead.